forte/processors/data_augment/algorithms/eda_ops.py

```diff
@@ -220,124 +219,143 @@
     Randomly choose two words in the sentence and swap their positions.
     Do this n times, where n = alpha * input length.
     """
 
-    def _augment(self, input_pack: MultiPack, aug_pack_names: List[str]):
+    def augment(self, data_pack: DataPack) -> bool:
         augment_entry = get_class(self.configs["augment_entry"])
         if not issubclass(augment_entry, Annotation):
             raise ValueError(
                 f"This augmenter only accept data of "
                 f"'forte.data.ontology.top.Annotation' type, "
                 f"but {self.configs['augment_entry']} is not."
             )
-        for pack_name in aug_pack_names:
-            data_pack: DataPack = input_pack.get_pack(pack_name)
-            annotations: List[Annotation] = list(data_pack.get(augment_entry))
+
+        try:
+            annotations: List[Annotation] = list(
+                data_pack.get(self.configs["augment_entry"])
+            )
             if len(annotations) > 0:
                 replace_map: Dict = {}
-                for _ in range(ceil(self.configs["alpha"] * len(annotations))):
-                    swap_idx = random.sample(range(len(annotations)), 2)
-                    new_idx_0 = (
-                        swap_idx[1]
-                        if swap_idx[1] not in replace_map
-                        else replace_map[swap_idx[1]]
-                    )
-                    new_idx_1 = (
-                        swap_idx[0]
-                        if swap_idx[0] not in replace_map
-                        else replace_map[swap_idx[0]]
-                    )
-                    replace_map[swap_idx[0]] = new_idx_0
-                    replace_map[swap_idx[1]] = new_idx_1
-                pid: int = data_pack.pack_id
-                for idx, replace_target in replace_map.items():
-                    self._replaced_annos[pid].add(
-                        (
-                            annotations[idx].span,
-                            annotations[replace_target].text,
-                        )
-                    )
+            for _ in range(ceil(self.configs["alpha"] * len(annotations))):
+                swap_idx = random.sample(range(len(annotations)), 2)
+                new_idx_0 = (
+                    swap_idx[1]
+                    if swap_idx[1] not in replace_map
+                    else replace_map[swap_idx[1]]
+                )
+                new_idx_1 = (
+                    swap_idx[0]
+                    if swap_idx[0] not in replace_map
+                    else replace_map[swap_idx[0]]
+                )
+                replace_map[swap_idx[0]] = new_idx_0
+                replace_map[swap_idx[1]] = new_idx_1
+
+            for idx, replace_target in replace_map.items():
+                self.replace_annotations(
+                    annotations[idx], annotations[replace_target].text
+                )
+            return True
+        except ValueError:
+            return False
```

Comment:
wait, not the `try` block still cover a very large piece of code. This could be problematic. Since you only intended to catch the `ValueError` from `replace_annotations`, you should only catch there.